src/mumble/Messages.cpp

```diff
@@ -59,6 +59,26 @@
 		return; \
 	}
 
+/// This message is being received after the client send a Capabilities message with its preferred protocols inside.
+/// The server replies with a Capabilities message with only one protocol in it to indicate the protocol it has chosen.
+///
+void MainWindow::msgCapabilities(const MumbleProto::Capabilities &msg) {
+	if (msg.supported_protocols_size() == 0)
+		return;
+
+	ConnectionPtr c = g.sh->cConnection;
+
+	auto protocol_type = msg.supported_protocols(0);
+	switch (protocol_type) {
+		case MumbleProto::Capabilities_VoiceProtocol_UDP_OCB2:
+			c->vptVoiceProtocolType = VoiceProtocolType::UDP_OCB2;
+			c->initializeCipher();
+		case MumbleProto::Capabilities_VoiceProtocol_CUSTOM:
+		default:
+			break;
```

Comment:
Client should disconnect here (and log an error)

Reply:
Perhaps we can keep the user logged in but display a message that he can't speak/hear because no available protocol can be used?

Reply:
Idk if it's that useful :thinking: 
I guess that everyone that uses Mumble via this client will actually want to talk...

Reply:
Ok. I think it makes sense to disconnect clients if no voice protocol is available. However, currently, even Opus initialization failure won't trigger the disconnection. (See https://github.com/mumble-voip/mumble/blob/73c535b25d1607131245f4969c638c686e506245/src/mumble/Messages.cpp#L1107)

I think we need to make this behavior consistent across the entire code base.